### PR TITLE
python[23]Packages: propagate overrides of python[23] to python[23]Pa…

### PR DESCRIPTION
**Pull request by SuperSandro2000**
…ckages

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
What is the rationale? Please add one to the commit message.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -15076,8 +15076,8 @@ with pkgs;
   python3 = python310;
 
   # pythonPackages further below, but assigned here because they need to be in sync
-  python2Packages = dontRecurseIntoAttrs python27Packages;
-  python3Packages = dontRecurseIntoAttrs python310Packages;
+  python2Packages = dontRecurseIntoAttrs python27.pkgs;
+  python3Packages = dontRecurseIntoAttrs python310.pkgs;
```
Comment (winterqt) [inline pkgs/top-level/all-packages.nix:15080]:
```suggestion
  python2Packages = dontRecurseIntoAttrs python2.pkgs;
  python3Packages = dontRecurseIntoAttrs python3.pkgs;
```

**Review by winterqt — CHANGES_REQUESTED** [review]
As is, this change does not fix the issue. I've tested this by attempting to build the following from the tree root:
```nix
(import ./. {
  overlays = [
    (self: super: {
      python3 = super.python3.override {
        packageOverrides = pyself: pysuper: {
          black = throw "test";
        };
      };
    })
  ];
}).black
```

This succeeds on master, as well as with this changed applied. With my suggestion applied, however, it fails as expected when overriding both `python3` and `python310`.

(P.S. Maybe we should move these attrs down to the other `pythonPackages` definitions now that they don't have to be in sync with the `python{2,3}` attributes?)

**Comment by SuperSandro2000** [comment]
Now your example throws.

**Comment by winterqt** [comment]
Why? This is still valid AFAIK.

**Pull request closed without merging**